src/components/MessagesList/MessagesList.vue

```diff
@@ -137,6 +142,24 @@ export default {
 	},
 
 	computed: {
+		isWindowVisible() {
+			return this.$store.getters.windowIsVisible() && this.isVisible
+		},
+
+		/**
+		 * Finds the first unread message component
+		 *
+		 * @returns {object} Vue component of the message or null if not found
+		 */
+		unreadMessageComponent() {
+			let el = document.getElementById('message_' + this.conversation.lastReadMessage)
+			if (el) {
+				el = el.closest('.message')
+			}
+
+			return el?.__vue__
```

Comment:
This looks illegal.

Reply:
Not too happy about it either.
I can work it around by storing the "seen" attribute also on the DOM to be able to access it from here.

Reply:
removed the aberration and replaced with DOM attributes "data-message-id" and "data-seen"